### PR TITLE
Pushing, sharing messages prototype

### PR DESCRIPTION
**Pull request by tomasvanagas**
Hello everyone,

I've finished what @rorpage started implementing - message pushing/sharing functionality.
Still it would be great to get input about naming, and general order of the code.

Hope we can fix these fast and merge into master :)

Screenshot 2020-07-10 at 12 50 56

Screenshot 2020-07-10 at 12 52 42

Screenshot 2020-07-10 at 12 51 58

### CODE REVIEW

File: app/src/main/java/com/github/gotify/sharing/ShareActivity.java [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+            appNameList.add(app.getName());
+        }
+        Spinner dynamicSpinner = findViewById(R.id.appSpinner);
+        ArrayAdapter<String> adapter =
+                new ArrayAdapter<String>(
+                        ShareActivity.this,
+                        android.R.layout.simple_spinner_dropdown_item,
+                        appNameList);
+        dynamicSpinner.setAdapter(adapter);
+        dynamicSpinner.setOnItemSelectedListener(
```
Comment (jmattheis) [inline app/src/main/java/com/github/gotify/sharing/ShareActivity.java:106]: Is this needed, seems like a noop implementation.

File: app/src/main/java/com/github/gotify/sharing/ShareActivity.java [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+        ApplicationHolder appsHolder = new ApplicationHolder(ShareActivity.this, client);
+        List<Application> apps = appsHolder.get();
+        appsHolder.onUpdate(() -> PopulateSpinner(appsHolder.get()));
+        appsHolder.request();
+
+        // Push button
+        final Button button = findViewById(R.id.button);
+        button.setOnClickListener(
+                new View.OnClickListener() {
+                    public void onClick(View v) {
```
Comment (jmattheis) [inline app/src/main/java/com/github/gotify/sharing/ShareActivity.java:62]: Extract this to a separate method and add a onClick handler with Butterknife:
```
    @OnClick(R.id.button)
    public void openDocumentation() {
         // content
    }
```

File: app/src/main/java/com/github/gotify/sharing/ShareActivity.java [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+        // App spinner
+        client =
+                ClientFactory.clientToken(settings.url(), settings.sslSettings(), settings.token());
+        ApplicationHolder appsHolder = new ApplicationHolder(ShareActivity.this, client);
+        List<Application> apps = appsHolder.get();
+        appsHolder.onUpdate(() -> PopulateSpinner(appsHolder.get()));
+        appsHolder.request();
+
+        // Push button
+        final Button button = findViewById(R.id.button);
```
Comment (jmattheis) [inline app/src/main/java/com/github/gotify/sharing/ShareActivity.java:59]: rename button to push_button and remove the comment above.

File: app/src/main/java/com/github/gotify/sharing/ShareActivity.java [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+            actionBar.setDisplayHomeAsUpEnabled(true);
+            actionBar.setDisplayShowCustomEnabled(true);
+        }
+        settings = new Settings(this);
+
+        // App spinner
+        client =
+                ClientFactory.clientToken(settings.url(), settings.sslSettings(), settings.token());
+        ApplicationHolder appsHolder = new ApplicationHolder(ShareActivity.this, client);
+        List<Application> apps = appsHolder.get();
```
Comment (jmattheis) [inline app/src/main/java/com/github/gotify/sharing/ShareActivity.java:54]: unused.

File: app/src/main/java/com/github/gotify/sharing/ShareActivity.java [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+
+                        Spinner appSpinner = findViewById(R.id.appSpinner);
+                        String selectedApp = appSpinner.getSelectedItem().toString();
+
+                        Spinner prioritySpinner = findViewById(R.id.prioritySpinner);
+                        Long priority =
+                                Long.parseLong(prioritySpinner.getSelectedItem().toString());
+                        message.setPriority(priority);
+
+                        new GetApps(selectedApp).execute(message);
```
Comment (jmattheis) [inline app/src/main/java/com/github/gotify/sharing/ShareActivity.java:81]: in GetApps the apps will be requested via network, use appsHolder.get() instead and pass the Application Object to the async task with the message.

File: app/src/main/java/com/github/gotify/sharing/ShareActivity.java [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+
+        @SuppressLint("WrongThread")
+        @Override
+        protected Void doInBackground(Message... messages) {
+            try {
+                ApplicationApi applicationApi = client.createService(ApplicationApi.class);
+                List<Application> apps = Api.execute(applicationApi.getApps());
+                for (Application app : apps) {
+                    if (app.getName().equals(SelectedApp)) {
+                        client =
```
Comment (jmattheis) [inline app/src/main/java/com/github/gotify/sharing/ShareActivity.java:140]: Don't reassign the client, instead pass the Application and Message to the AsyncTask and then create a client just for the request.

**Comment by tomasvanagas** [comment]
Hey @jmattheis ,

Thanks for the fast review!
I've made the changes, please review my poor code :)

And sorry for my messy github usage, still learning :)

Jul-12-2020 03-43-20

Sharing

**Comment by jmattheis** [comment]
I've made some more changes to use material ui text views. (I've made the changes because I didn't knew exactly what I wanted, so I tinkered a little). Also, I cleaned up some minor stuff. Can you look over the changes I made?

**Comment by tomasvanagas** [comment]
Checked, looks great! :)

**Comment by jmattheis** [comment]
Great. I've cleaned up the commit history to remove the pr commits. Thanks for your contribution!

**Pull request merged**